pymongo/asynchronous/mongo_client.py

```diff
@@ -1173,6 +1175,22 @@ def __getitem__(self, name: str) -> database.AsyncDatabase[_DocumentType]:
         """
         return database.AsyncDatabase(self, name)
 
+    def __del__(self) -> None:
+        """Check that this AsyncMongoClient has been closed and issue a warning if not."""
+        # TODO: Remove in https://jira.mongodb.org/browse/PYTHON-4731
```

Comment:
I didn't realize we removed `MongoClient.__del__`.  I think we should preserver the 4.8 behavior and change `connect=False` and remove the `__del__` behavior together in 5.0

Reply:
Wouldn't we want to keep the `__del__` warning forever?

Reply:
Oh wait, we never had a `__del__`.  Yes, I agree.

Reply:
We can remove this comment now.